### DISCUSSION TITLE
GY-521/MPU-6050 Alternative?

### DISCUSSION THREAD

**Discussion by mgbasi** [universam1/iSpindel#540] [Ideas]
It looks like—albeit with 10s or 100s of thousands left in stock—that the MPU-6050 chip is discontinued. Are there plans to make an alternative iSpindel firmware variation that uses a newer gyroscopic module?

**Comment by Bananamannn** [comment]
I defiantly hope so. I just received 20 MPU-6500 boards that I accidently ordered instead of 6050s for a batch of ISpindels I'm building for my homebrew club. 
I don't know enough about platfrom.io to change the source code from 6050 to 6500 :(

Any help with that would be greatly appreciated.

**Reply by thegreatgunbantoad** [reply]
It's definitely worth building with a PCB like the Cherry Philip one.

**Reply by Bananamannn** [reply]
I have used the 6500 with the 6050 library before and it appears to work just fine, the only thing i've found that doesn't work is the .isconnected function as the address is different. I have done the below code in the past to make my own connected test and it's worked well. I don't knowhow to do pull requests or anything though

```
bool testAccel()
{
  uint8_t res = Wire.status();
    if (res != I2C_OK)
    CONSOLELN(String(F("I2C ERROR: ")) + res);
  
  bool con = false;
  if (accelgyro.getDeviceID()==0x34)
  {
    con = true;
  }
  
  if (accelgyro.getDeviceID()==0x38)
  {
    con = true;
  }
  
  if (!con)
    CONSOLELN(F("Acc Test Connection ERROR!"));
      
  return res == I2C_OK && con == true;
}
```
